### ISSUE TITLE
Using GitHub with GITHUB_TOKEN without Prefix leaks token

### ISSUE THREAD

**Issue by rowi1de** [octokit/octokit.js#1517]
```ts
import * as core from '@actions/core';
import { context, GitHub } from '@actions/github'

//comes from {{secrets.GITHUB_TOKEN}}
const repoToken = core.getInput('repo-token', { required: true })
const client = new GitHub(repoToken)
```

will leak the token on error: 
> "Could not resolve to a node with the global id of 'M....c='."

```bash
19-11-22T16:44:34.9046342Z (node:2404) UnhandledPromiseRejectionWarning: HttpError: Validation Failed: "Could not resolve to a node with the global id of 'M....c='."
2019-11-22T16:44:34.9047036Z     at /home/runner/work/_actions/rowi1de/typescript-action/master/node_modules/@octokit/request/dist-node/index.js:66:23
2019-11-22T16:44:34.9047273Z     at processTicksAndRejections (internal/process/task_queues.js:89:5)
201
```

see https://github.com/actions/typescript-action/issues/40

**Comment by gr2m** [comment]
Hmmm can you manage to reproduce the problem with just using `@octokit/rest`? 

I would recommend to use [`@octokit/action`](https://github.com/octokit/action.js). You need to set the `GITHUB_TOKEN` environment variable as it's the convention, it will take care of authorization for you.

It does not have all the named methods for all the REST API endpoints yet, but you have `octokit.graphql` and `octokit.request` which is usually all that you need.

**Comment by rowi1de** [comment]
I’m passing `{{secrets.GITHUB_TOKEN}}` as the parameter into my action.
This value is set by GitHub automatically, as documented here https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#using-the-github_token-in-a-workflow

Using the action you mentioned could be an alternative. But not sure if that would make a difference 


In the second example (direct rest call) they ass the bearer prefix: https://help.github.com/en/actions/automating-your-workflow-with-github-actions/authenticating-with-the-github_token#example-calling-the-rest-api 

Do I need to tell the Oktokit client what kind of token it is?

**Comment by gr2m** [comment]
I can't tell what exactly is happening, but I don't think that the problem is with `@octokit/rest`, but with `@actions/github`. I did not write the code for the latter and I don't know if it's maintained. 

I've added the REST API endpoint methods to `@octokit/action` yesterday, so it should be simple to use that now, and it will take care of authentication for you, without the need to manually pass in a token, as long as you set the `GITHUB_TOKEN` environment variable.
